src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,141 @@
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    private static User defaultUser;
+    private static final int MIN_AGE = 18;
+    private static final String VALID_PASSWORD = "123456";
+    private static final String DEFAULT_LOGIN = "testUser";
+
+    @BeforeAll
+    static void initializeService() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void initialUser() {
+        defaultUser = new User(DEFAULT_LOGIN, VALID_PASSWORD, MIN_AGE);
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void register_CorrectUser_Ok() {
+        User actual = registrationService.register(defaultUser);
+        boolean cont = Storage.people.contains(actual);
+        assertTrue(cont, "User should be add to Storage");
+    }
+
+    @Test
+    void register_addNullUser_NotOk() {
+        User user = null;
+        assertThrows(InvalidUserException.class,
+                () -> registrationService.register(user),
+                "User can't be null");
+    }
+
+    @Test
+    void register_addNullLogin_NotOk() {
+        defaultUser.setLogin(null);
+        assertThrows(InvalidUserException.class, () -> registrationService.register(defaultUser),
+                "Login can't be null, please fixed this. Login: "
+                        + defaultUser.getLogin());
+    }
+
+    @Test
+    void register_addNullPassword_NotOk() {
+        defaultUser.setPassword(null);
+        assertThrows(InvalidUserException.class, () -> registrationService.register(defaultUser),
+                "Password can't be null, please fixed this. password = "
+                        + defaultUser.getPassword());
+    }
+
+    @Test
+    void register_passwordLengthIsLessThanMinLength_NotOk() {
+        defaultUser.setPassword("12345");
+        assertThrows(InvalidUserException.class, () ->
+                        registrationService.register(defaultUser),
+                "Password length should be least 6 characters, but length "
+                        + defaultUser.getPassword().length());
+    }
+
+    @Test
+    void register_passwordLengthMinLength_Ok() {
+        defaultUser.setPassword("123456");
+        registrationService.register(defaultUser);
+        boolean actual = Storage.people.contains(defaultUser);
+        assertTrue(actual, "Password length: "
+                + defaultUser.getPassword().length()
+                + ". User should be add to Storage");
+    }
+
+    @Test
+    void register_passwordLengthIsGreaterThanMinLength_Ok() {
+        defaultUser.setPassword("1234567");
+        registrationService.register(defaultUser);
+        boolean checkUser = Storage.people.contains(defaultUser);
```

Comment:
```suggestion
        boolean isExists = Storage.people.contains(defaultUser);
```
fix everywhere naming of boolean variables